include/nil/blueprint/assigner.hpp

```diff
@@ -1495,84 +1461,68 @@ namespace nil {
+                                                        gep->getSourceElementType(), gep_indices, frame);
                             if (gep_res == 0) {
                                 std::cerr << "Incorrect GEP result!" << std::endl;
                                 return nullptr;
                             }
                             std::ostringstream oss;
                             oss << gep_res.data;
                             log.debug(boost::format("GEP: %1%") % oss.str());
-                            frame.scalars[gep] = put_into_assignment(gep_res);
+                            frame.scalars[gep] = save_value(gep_res);
                         } else {
                             log.debug(boost::format("Skip GEP"));
-                            frame.scalars[gep] = put_into_assignment(0);
+                            frame.scalars[gep] = save_value(0);
                         }
                         return inst->getNextNonDebugInstruction();
                     }
                     case llvm::Instruction::Load: {
                         auto *load_inst = llvm::cast<llvm::LoadInst>(inst);
-                        ptr_type ptr = 0;
-                        if (gen_mode.has_assignments()) {
-                            ptr = resolve_number<ptr_type>(frame, load_inst->getPointerOperand());
-                        }
+                        ptr_type ptr = resolve_number<ptr_type>(frame, load_inst->getPointerOperand());
                         log.debug(boost::format("Load: %1%") % ptr);
                         handle_load(ptr, load_inst, frame);
                         return inst->getNextNonDebugInstruction();
                     }
                     case llvm::Instruction::Store: {
+                        auto *store_inst = llvm::cast<llvm::StoreInst>(inst);
+                        ptr_type ptr = resolve_number<ptr_type>(frame, store_inst->getPointerOperand());
                         if (gen_mode.has_assignments()) {
-                            auto *store_inst = llvm::cast<llvm::StoreInst>(inst);
-                            ptr_type ptr = resolve_number<ptr_type>(frame, store_inst->getPointerOperand());
                             log.debug(boost::format("Store: %1%") % ptr);
                             const llvm::Value *val = store_inst->getValueOperand();
                             handle_store(ptr, val, frame);
                         } else {
-                            log.debug(boost::format("Skip Store"));
+                            log.debug(boost::format("Skip store: %1%") % ptr);
                         }
                         return inst->getNextNonDebugInstruction();
                     }
                     case llvm::Instruction::InsertValue: {
                         auto *insert_inst = llvm::cast<llvm::InsertValueInst>(inst);
+                        ptr_type ptr = resolve_number<ptr_type>(frame, insert_inst->getAggregateOperand());
```

Comment:
This comment applies to this code, but it's a general one actually: some of these changes will have a hard time rebasing on new memory model. I don't know what to do with that.